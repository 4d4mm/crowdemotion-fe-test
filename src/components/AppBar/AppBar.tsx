import * as React from "react";
import { Link } from 'react-router-dom';
import { makeStyles } from "@material-ui/core";
import { AppBar, Toolbar, Typography } from "@material-ui/core";

const useStyles = makeStyles(
  (theme) => ({
    link: {
      textDecoration: "none",
      display: "flex",
      alignItems: "center"
    },
    title: {
      fontFamily: "'Comfortaa', cursive",
      letterSpacing: "-1px",
      color: theme.palette.secondary.contrastText,
    }
  }),
  {
    name: "CrowdEmotionAppBar"
  }
);

type CrowdEmotionAppBarProps = {
  logo: React.ComponentType<React.ComponentProps<any>>;
};

const CrowdEmotionAppBar: React.SFC<CrowdEmotionAppBarProps> = ({
  logo: Logo
}) => {
  const { title, link } = useStyles({});
  return (
    <AppBar position="fixed" color="secondary">
      <Toolbar variant="dense">
        <Link to="/" className={link}>
          <Logo />
          &nbsp;
          <Typography variant="h5" className={title}>
            <b>crowd</b>emotion
          </Typography>
        </Link>
      </Toolbar>
    </AppBar>
  );
};

export default CrowdEmotionAppBar;
