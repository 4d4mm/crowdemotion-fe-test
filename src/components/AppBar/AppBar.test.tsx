import React from "react";
import { MemoryRouter } from "react-router-dom";
import { mount } from "enzyme";
import AppBar from "./AppBar";

const MockLogo: React.SFC = () => <div>logo</div>;

describe("AppBar", () => {
  it("renders button with default ui", () => {
    const renderedAppBar = mount(<MemoryRouter initialEntries={[{pathname:'/', key:'test'}]}><AppBar logo={MockLogo} /></MemoryRouter>);
    expect(renderedAppBar).toMatchSnapshot();
  });
});
