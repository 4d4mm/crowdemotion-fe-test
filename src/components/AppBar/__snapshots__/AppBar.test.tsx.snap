// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders button with default ui 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "test",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "test",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "test",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CrowdEmotionAppBar
      logo={[Function]}
    >
      <WithStyles(ForwardRef(AppBar))
        color="secondary"
        position="fixed"
      >
        <ForwardRef(AppBar)
          classes={
            Object {
              "colorDefault": "MuiAppBar-colorDefault",
              "colorPrimary": "MuiAppBar-colorPrimary",
              "colorSecondary": "MuiAppBar-colorSecondary",
              "positionAbsolute": "MuiAppBar-positionAbsolute",
              "positionFixed": "MuiAppBar-positionFixed",
              "positionRelative": "MuiAppBar-positionRelative",
              "positionStatic": "MuiAppBar-positionStatic",
              "positionSticky": "MuiAppBar-positionSticky",
              "root": "MuiAppBar-root",
            }
          }
          color="secondary"
          position="fixed"
        >
          <WithStyles(ForwardRef(Paper))
            className="MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary mui-fixed"
            component="header"
            elevation={4}
            square={true}
          >
            <ForwardRef(Paper)
              className="MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary mui-fixed"
              classes={
                Object {
                  "elevation0": "MuiPaper-elevation0",
                  "elevation1": "MuiPaper-elevation1",
                  "elevation10": "MuiPaper-elevation10",
                  "elevation11": "MuiPaper-elevation11",
                  "elevation12": "MuiPaper-elevation12",
                  "elevation13": "MuiPaper-elevation13",
                  "elevation14": "MuiPaper-elevation14",
                  "elevation15": "MuiPaper-elevation15",
                  "elevation16": "MuiPaper-elevation16",
                  "elevation17": "MuiPaper-elevation17",
                  "elevation18": "MuiPaper-elevation18",
                  "elevation19": "MuiPaper-elevation19",
                  "elevation2": "MuiPaper-elevation2",
                  "elevation20": "MuiPaper-elevation20",
                  "elevation21": "MuiPaper-elevation21",
                  "elevation22": "MuiPaper-elevation22",
                  "elevation23": "MuiPaper-elevation23",
                  "elevation24": "MuiPaper-elevation24",
                  "elevation3": "MuiPaper-elevation3",
                  "elevation4": "MuiPaper-elevation4",
                  "elevation5": "MuiPaper-elevation5",
                  "elevation6": "MuiPaper-elevation6",
                  "elevation7": "MuiPaper-elevation7",
                  "elevation8": "MuiPaper-elevation8",
                  "elevation9": "MuiPaper-elevation9",
                  "root": "MuiPaper-root",
                  "rounded": "MuiPaper-rounded",
                }
              }
              component="header"
              elevation={4}
              square={true}
            >
              <header
                className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary mui-fixed"
              >
                <WithStyles(ForwardRef(Toolbar))
                  variant="dense"
                >
                  <ForwardRef(Toolbar)
                    classes={
                      Object {
                        "dense": "MuiToolbar-dense",
                        "gutters": "MuiToolbar-gutters",
                        "regular": "MuiToolbar-regular",
                        "root": "MuiToolbar-root",
                      }
                    }
                    variant="dense"
                  >
                    <div
                      className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
                    >
                      <Link
                        className="CrowdEmotionAppBar-link-1"
                        to="/"
                      >
                        <LinkAnchor
                          className="CrowdEmotionAppBar-link-1"
                          href="/"
                          navigate={[Function]}
                        >
                          <a
                            className="CrowdEmotionAppBar-link-1"
                            href="/"
                            onClick={[Function]}
                          >
                            <MockLogo>
                              <div>
                                logo
                              </div>
                            </MockLogo>
                            Â 
                            <WithStyles(ForwardRef(Typography))
                              className="CrowdEmotionAppBar-title-2"
                              variant="h5"
                            >
                              <ForwardRef(Typography)
                                className="CrowdEmotionAppBar-title-2"
                                classes={
                                  Object {
                                    "alignCenter": "MuiTypography-alignCenter",
                                    "alignJustify": "MuiTypography-alignJustify",
                                    "alignLeft": "MuiTypography-alignLeft",
                                    "alignRight": "MuiTypography-alignRight",
                                    "body1": "MuiTypography-body1",
                                    "body2": "MuiTypography-body2",
                                    "button": "MuiTypography-button",
                                    "caption": "MuiTypography-caption",
                                    "colorError": "MuiTypography-colorError",
                                    "colorInherit": "MuiTypography-colorInherit",
                                    "colorPrimary": "MuiTypography-colorPrimary",
                                    "colorSecondary": "MuiTypography-colorSecondary",
                                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                    "displayBlock": "MuiTypography-displayBlock",
                                    "displayInline": "MuiTypography-displayInline",
                                    "gutterBottom": "MuiTypography-gutterBottom",
                                    "h1": "MuiTypography-h1",
                                    "h2": "MuiTypography-h2",
                                    "h3": "MuiTypography-h3",
                                    "h4": "MuiTypography-h4",
                                    "h5": "MuiTypography-h5",
                                    "h6": "MuiTypography-h6",
                                    "noWrap": "MuiTypography-noWrap",
                                    "overline": "MuiTypography-overline",
                                    "paragraph": "MuiTypography-paragraph",
                                    "root": "MuiTypography-root",
                                    "srOnly": "MuiTypography-srOnly",
                                    "subtitle1": "MuiTypography-subtitle1",
                                    "subtitle2": "MuiTypography-subtitle2",
                                  }
                                }
                                variant="h5"
                              >
                                <h5
                                  className="MuiTypography-root CrowdEmotionAppBar-title-2 MuiTypography-h5"
                                >
                                  <b>
                                    crowd
                                  </b>
                                  emotion
                                </h5>
                              </ForwardRef(Typography)>
                            </WithStyles(ForwardRef(Typography))>
                          </a>
                        </LinkAnchor>
                      </Link>
                    </div>
                  </ForwardRef(Toolbar)>
                </WithStyles(ForwardRef(Toolbar))>
              </header>
            </ForwardRef(Paper)>
          </WithStyles(ForwardRef(Paper))>
        </ForwardRef(AppBar)>
      </WithStyles(ForwardRef(AppBar))>
    </CrowdEmotionAppBar>
  </Router>
</MemoryRouter>
`;
